java/memory/src/test/java/org/apache/arrow/memory/util/TestArrowBufPointer.java

```diff
@@ -67,4 +67,29 @@ public void testArrowBufPointersEqual() {
       }
     }
   }
+
+  @Test
+  public void testArrowBufPointersHashCode() {
+    final int vectorLength = 100;
+    try (ArrowBuf buf1 = allocator.buffer(vectorLength * 4);
+         ArrowBuf buf2 = allocator.buffer(vectorLength * 4)) {
+      for (int i = 0; i < vectorLength; i++) {
+        buf1.setInt(i * 4, i);
+        buf2.setInt(i * 4, i);
+      }
+
+      ArrowBufPointer pointer1 = new ArrowBufPointer();
```

Comment:
Please separate these tests into null and not-null.  Also, it would be nice to have a test to verify caching.

Reply:
Good suggestion.
I have added test cases to 1) verify cases for null; 2) verify the hash code is recomputed/reused.